src/register.js

```diff
@@ -13,24 +13,30 @@ class Register {
 
   /**
    * Find services in the register that conform to a profile
-   * 
+   *
    * @param {String} profileId Id of the profile the matching sarvice has to support
    */
   async findServices(profileId) {
-    const response =
-      await superagent
-        .get(`${this.registerURL}/search/`)
-        .query({ semanticProfile: profileId })
-        .set('accept', 'application/json')
-
-    const services = response.body['disco'];
-    if (!services || !services.length)
-      return Promise.reject(`No service for profile '${profileId}' found.`);
-
-    return services;
+    try {
+      const response = await fetch(
+        `${this.registerURL}/search?semanticProfile=${encodeURIComponent(profileId)}`,
```

Comment:
@schovi any good cross (node.js and browser) library that works with URL? (https://developer.mozilla.org/en-US/docs/Web/API/URL/URL)